src/components/ViewDate.jsx

```diff
@@ -19,17 +19,23 @@ class ViewDate extends React.Component {
     this.state = { open: false }
   }
 
-  weekNum () {
-    return moment(this.props.viewDate).format('Wo')
-  }
+  weekParity (weekParity) {
+    if (weekParity !== 'even' && weekParity !== 'odd') {
+      return '-'
+    }
 
-  weekParity () {
-    const parity = moment(this.props.viewDate).isoWeek() % 2 === 0 ? 'even' : 'odd'
-    return CP.translate(parity)
+    return CP.translate(weekParity)
   }
 
   render () {
-    return <div className="view-date">{CP.translate('weekNav.teachweek', { weeknum: this.weekNum() })} ({this.weekParity()})</div>
+    const weekParity = this.weekParity(this.props.weekParity)
+    const weekText = CP.translate('weekNav.week_' + this.props.weekType, { weeknum: this.props.weekNum, parity: weekParity })
+    const translationExists = ['teaching', 'exams', 'holiday'].indexOf(this.props.weekType) >= 0
```

Comment:
Well, the condition order doesn't matter, because CP.translate catches the exception and returns string, something like 'the key blah blah doesn't exists'. But I agree, when you don't know this, it doesn't make sense. Fixed.